### PR TITLE
added st_makeenvelope builtin

### LINKED ISSUE #80357: builtins: implement ST_MakeEnvelope(float xmin, float ymin, float xmax, float ymax, integer srid=unknown)
**Is your feature request related to a problem? Please describe.**
Our app makes extensive use of geometry bounds (e.g. to filter geo-located things).

We use the [ST_MakeEnvelope](https://postgis.net/docs/ST_MakeEnvelope.html) function to build geometries suitable for geospatial indexes out.
```
ST_MakeEnvelope(float xmin, float ymin, float xmax, float ymax, integer srid=unknown) -> geometry
``` 

But this is not supported by cockroachDB as v22.1.x.

**Describe the solution you'd like**
Have support for `ST_MakeEnvelope(loat xmin, float ymin, float xmax, float ymax, integer srid=unknown) -> geometry`

**Describe alternatives you've considered**
As of v22.1.x, we only have the following alternative:
* either build a geometry in the client code from the bounds then pass it as a geometry arg to the query
* or use the following hack: ` ST_SetSRID(ST_Envelope(ST_Makebox2d(ST_Makepoint(?,?), ST_Makepoint(?,?))), 4326)`

**Additional context**
N/A

----

You'll need to implement both variants to make this work>
```
ST_MakeEnvelope(float xmin, float ymin, float xmax, float ymax)
ST_MakeEnvelope(float xmin, float ymin, float xmax, float ymax, integer srid=unknown)
```

See #48814 for example PR

### PR DESCRIPTION
**Pull request by fredbi**
* fixes #80357

Release note (sql change): added builtin st_makeenvelope

Signed-off-by: Frédéric BIDON <frederic@oneconcern.com>

### CODE REVIEW

File: pkg/sql/sem/builtins/geo_builtins.go [hunk 1]
```diff
@@ -5558,6 +5558,42 @@ Bottom Left.`,
...
+				{"xmax", types.Float},
+				{"ymax", types.Float},
+				{"srid", types.Int},
+			},
+			ReturnType: tree.FixedReturnType(types.Geometry),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				return makeGeometryFromBounds(args)
+			},
+			Info: infoBuilder{
+				info: "Returns a box2d from bounds.",
```
Comment (otan) [inline pkg/sql/sem/builtins/geo_builtins.go:5576]: nit: this comment is incorrect.
"Creates a rectangular Polygon from the minimum and maximum values for X and Y with the given SRID."

File: pkg/sql/sem/builtins/geo_builtins.go [hunk 2]
```diff
@@ -5558,6 +5558,42 @@ Bottom Left.`,
...
+				{"ymin", types.Float},
+				{"xmax", types.Float},
+				{"ymax", types.Float},
+			},
+			ReturnType: tree.FixedReturnType(types.Geometry),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				return makeGeometryFromBounds(args)
+			},
+			Info: infoBuilder{
+				info: "Returns a box2d from bounds (the default srid value is 0).",
```
Comment (otan) [inline pkg/sql/sem/builtins/geo_builtins.go:5592]: "Creates a rectangular Polygon from the minimum and maximum values for X and Y with SRID 0."

File: pkg/sql/sem/builtins/geo_builtins_test.go [hunk 3]
```diff
@@ -103,3 +104,88 @@ func TestGeoBuiltinsPointEmptyArgs(t *testing.T) {
 		})
 	}
 }
+
+func mustParseDFloat(t testing.TB, in string) tree.Datum {
```
Comment (otan) [inline pkg/sql/sem/builtins/geo_builtins_test.go:108]: please remove these tests and add it as a logic test to https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial instead. use `make testbaselogic FILES='geospatial'  TESTFLAGS='-rewrite'` to rewrite the output and verify it is correct

File: pkg/sql/sem/builtins/geo_builtins.go [hunk 4]
```diff
@@ -7502,3 +7538,28 @@ func applyGeoindexConfigStorageParams(
 	}
 	return indexDesc.GeoConfig, nil
 }
+
+// makeGeometryFromBounds builds a rectangle geometry from types.Float bounds as datums
+func makeGeometryFromBounds(args tree.Datums) (tree.Datum, error) {
```
Comment (otan) [inline pkg/sql/sem/builtins/geo_builtins.go:7543]: nit: rename as `stEnvelopeFromArgs`

**Review by otan — COMMENTED** [review]
thanks! just some small suggestions

**Review by fredbi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80408)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @fredbi and @otan)

___
*[`pkg/sql/sem/builtins/geo_builtins.go` line 5592 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/80408#-N0I0P8-IwJLWZErKE1k-r1-5592:-N0kIeFw98pg43V-_n5g:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/35f2510bab74ee3a2080c775f1f65383dc9f3181/pkg/sql/sem/builtins/geo_builtins.go#L5592)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

"Creates a rectangular Polygon from the minimum and maximum values for X and Y with SRID 0."
</blockquote></details>

Done.

**Review by fredbi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80408)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[`pkg/sql/sem/builtins/geo_builtins_test.go` line 108 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/80408#-N0I0P8-IwJLWZErKE1l-r1-108:-N0klm7PD9cw7MC3gJld:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/35f2510bab74ee3a2080c775f1f65383dc9f3181/pkg/sql/sem/builtins/geo_builtins_test.go#L108)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

please remove these tests and add it as a logic test to https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial instead. use `make testbaselogic FILES='geospatial'  TESTFLAGS='-rewrite'` to rewrite the output and verify it is correct
</blockquote></details>

Done.

**Comment by fredbi** [comment]
@otan I believe I've addressed all your comments for this change.

**Review by otan — APPROVED** [review]
thanks!

squashed to one commit

bors r+

**Pull request merged**